dotnet/samples/GettingStartedWithVectorStores/Step4_NonStringKey_VectorStore.cs

```diff
@@ -1,195 +1,198 @@
-        {
-            return this._collection.DeleteCollectionAsync(cancellationToken);
-        }
-
-        /// <inheritdoc />
-        public async Task<TPublicRecord?> GetAsync(TPublicKey key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
-        {
-            var internalRecord = await this._collection.GetAsync(this._publicToInternalKeyMapper(key), options, cancellationToken).ConfigureAwait(false);
-            if (internalRecord == null)
-            {
-                return default;
-            }
-
-            return this._internalToPublicRecordMapper(internalRecord);
-        }
-
-        /// <inheritdoc />
-        public IAsyncEnumerable<TPublicRecord> GetBatchAsync(IEnumerable<TPublicKey> keys, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
-        {
-            var internalRecords = this._collection.GetBatchAsync(keys.Select(this._publicToInternalKeyMapper), options, cancellationToken);
-            return internalRecords.Select(this._internalToPublicRecordMapper);
-        }
-
-        /// <inheritdoc />
-        public async Task<TPublicKey> UpsertAsync(TPublicRecord record, CancellationToken cancellationToken = default)
-        {
-            var internalRecord = this._publicToInternalRecordMapper(record);
-            var internalKey = await this._collection.UpsertAsync(internalRecord, cancellationToken).ConfigureAwait(false);
-            return this._internalToPublicKeyMapper(internalKey);
-        }
-
-        /// <inheritdoc />
-        public async IAsyncEnumerable<TPublicKey> UpsertBatchAsync(IEnumerable<TPublicRecord> records, [EnumeratorCancellation] CancellationToken cancellationToken = default)
-        {
-            var internalRecords = records.Select(this._publicToInternalRecordMapper);
-            var internalKeys = this._collection.UpsertBatchAsync(internalRecords, cancellationToken);
-            await foreach (var internalKey in internalKeys.ConfigureAwait(false))
-            {
-                yield return this._internalToPublicKeyMapper(internalKey);
-            }
-        }
-
-        /// <inheritdoc />
-        public async Task<VectorSearchResults<TPublicRecord>> VectorizedSearchAsync<TVector>(TVector vector, VectorSearchOptions? options = null, CancellationToken cancellationToken = default)
-        {
-            var searchResults = await this._collection.VectorizedSearchAsync(vector, options, cancellationToken).ConfigureAwait(false);
-            var publicResultRecords = searchResults.Results.Select(result => new VectorSearchResult<TPublicRecord>(this._internalToPublicRecordMapper(result.Record), result.Score));
-
-            return new VectorSearchResults<TPublicRecord>(publicResultRecords)
-            {
-                TotalCount = searchResults.TotalCount,
-                Metadata = searchResults.Metadata,
-            };
-        }
-    }
-}
+// TODO: See note in MappingVectorStoreRecordCollection
```

Comment:
Should we just delete it or add Obsolete?

Reply:
Note that this isn't part of the public API - it's just a sample - so I'm not sure obsoleting makes sense. Also, obsoletion is generally for something that still works but is no longer the recommended way, whereas this no longer works; or rather, it can be made to work for the legacy/obsolete filter, but not (easily) for the new one.

@westey-m any opinion on what to do here? We can also leave commented out for now and deal with it later.